### ISSUE TITLE
用infer_det.py测试检测效果错误

### ISSUE THREAD

**Issue by suchxyz** [PaddlePaddle/PaddleOCR#5319]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：window、 python3.9

- 版本号/Version：Paddle：2.2.1  PaddleOCR： 2.4
 
-  问题相关组件/Related components：

- 运行指令/Command Code：  
python3 PaddleOCR/tools/infer_det.py -c ./configs/det/det_mv3_db.yml -o Global.infer_img=".\doc\imgs_en\img_10.jpg" Global.pretrained_model=".\pretrain_models\MobileNetV3_large_x0_5_pretrained"

- 完整报错/Complete Error Message：

按照教程https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.4/doc/doc_ch/detection.md 
**使用预训练模型， 测试单张图像的检测效果，结果都在图像边缘，而且都是很小的矩形框。所有的图片都是如此。 
另外执行多次上述命令，每次结果不固定，而且经常没有结果**

img_10.jpg分别率为1280*720
具体结果如下：
.\doc\imgs_en\img_10.jpg	[{"transcription": "", "points": [[1121, 716], [1126, 711], [1131, 716], [1126, 720]]}, {"transcription": "", "points": [[1119, 714], [1125, 714], [1125, 719], [1119, 719]]}, {"transcription": "", "points": [[391, 711], [397, 717], [392, 720], [387, 716]]}, {"transcription": "", "points": [[94, 713], [106, 713], [106, 720], [94, 720]]}, {"transcription": "", "points": [[324, 710], [331, 710], [331, 717], [324, 717]]}, {"transcription": "", "points": [[400, 713], [405, 708], [410, 713], [405, 718]]}, {"transcription": "", "points": [[1237, 704], [1243, 710], [1239, 715], [1232, 708]]}]

配置文件./configs/det/det_mv3_db.yml如下： 只改了用cpu跑
Global:
  use_gpu: false
  epoch_num: 50
  log_smooth_window: 20
  print_batch_step: 2
  save_model_dir: ./output/db_mv3/
  save_epoch_step: 2
  
  eval_batch_step: [0, 100]
  cal_metric_during_train: False
  pretrained_model: ./pretrain_models/MobileNetV3_large_x0_5_pretrained
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./output/det_db/predicts_db.txt

Architecture:
  model_type: det
  algorithm: DB
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: DBFPN
    out_channels: 256
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/train_icdar2015_label.txt
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
            - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
      - EastRandomCropData:
          size: [640, 640]
          max_tries: 50
          keep_ratio: true
      - MakeBorderMap:
          shrink_ratio: 0.4
          thresh_min: 0.3
          thresh_max: 0.7
      - MakeShrinkMap:
          shrink_ratio: 0.4
          min_text_size: 8
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 16
    num_workers: 8
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/icdar2015/text_localization/
    label_file_list:
      - ./train_data/icdar2015/text_localization/test_icdar2015_label.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
          image_shape: [736, 1280]
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 8
    use_shared_memory: False

另外使用自带模型，直接检测，没有任何问题
python .\tools\infer\predict_det.py --image_dir=".\doc\imgs_en\img_10.jpg" --det_model_dir=".\inference\ch_PP-OCRv2_det_infer\"

**Comment by littletomatodonkey** [comment]
你这个预训练模型不对，infer det.py需要加载训练好的模型，可以从repo首页下载

**Comment by suchxyz** [comment]
后面用的ch_ppocr_server_v2.0_det_train模型，配置文件用的det_res18_db_v2.0.yml， 可以正常训练

https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.4/doc/doc_ch/models_list.md
![image](https://user-images.githubusercontent.com/51903573/151094272-4de6dd34-8680-44f6-ac9d-06f054a22323.png)

**Comment by suchxyz** [comment]
> 你这个预训练模型不对，infer det.py需要加载训练好的模型，可以从repo首页下载

现在检测模型训练正常了。
但是找不到一个对的**中文识别**训练模型。试了以下3个模型
![image](https://user-images.githubusercontent.com/51903573/151094660-7f407212-2a1c-42e4-910e-7d9f98fd517a.png)
![image](https://user-images.githubusercontent.com/51903573/151094680-dba8358b-e02e-4892-b2ed-10aab8da040b.png)

**infer_rec.py的时候结果都不对：**  
[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] !
[2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] !
[2022/01/25 18:36:17] root INFO: load pretrain successful from pretrain_models/ch_ppocr_mobile_v2.0_rec_train/best_accuracy
[2022/01/25 18:36:17] root INFO: infer_img: doc/imgs_words/ch/word_1.jpg
[2022/01/25 18:36:17] root INFO: result: 尬棍妗诺喜５瀑痧群瀑Ω,嗔當恐尬帜尬寻践妗程意飼践醚践还 0.00046281534
[2022/01/25 18:36:17] root INFO: success!

btw：英文的随便找了个rec_mv3_none_bilstm_ctc_v2.0_train，是可以的。

求问，这个是怎么回事？

**Comment by suchxyz** [comment]
问题找到了， 是改了字典，导致结果不对

**Comment by suchxyz** [comment]
> > 你这个预训练模型不对，infer det.py需要加载训练好的模型，可以从repo首页下载
> 
> 现在检测模型训练正常了。 但是找不到一个对的**中文识别**训练模型。试了以下3个模型 ![image](https://user-images.githubusercontent.com/51903573/151094660-7f407212-2a1c-42e4-910e-7d9f98fd517a.png)
> 
> **infer_rec.py的时候结果都不对：** [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.weight [96, 6627] not matched with loaded params head.fc.weight [96, 6625] ! [2022/01/25 18:36:17] root WARNING: The shape of model params head.fc.bias [6627] not matched with loaded params head.fc.bias [6625] ! [2022/01/25 18:36:17] root INFO: load pretrain successful from pretrain_models/ch_ppocr_mobile_v2.0_rec_train/best_accuracy [2022/01/25 18:36:17] root INFO: infer_img: doc/imgs_words/ch/word_1.jpg [2022/01/25 18:36:17] root INFO: result: 尬棍妗诺喜５瀑痧群瀑Ω,嗔當恐尬帜尬寻践妗程意飼践醚践还 0.00046281534 [2022/01/25 18:36:17] root INFO: success!
> 
> btw：英文的随便找了个rec_mv3_none_bilstm_ctc_v2.0_train，是可以的。
> 
> 求问，这个是怎么回事？

问题找到了， 是改了字典，导致结果不对

**Comment by hlee13** [comment]
> 

只能加载预训练么？
不能从0自己训练？还是需要特别久呢
